centml/cli/cluster.py

```diff
@@ -20,10 +21,46 @@ def convert(self, value, param, ctx):
             return None  # to avoid warning from lint for inconsistent return statements
 
 
-hw_to_id_map = {"small": 1000, "medium": 1001, "large": 1002}
-id_to_hw_map = {v: k for k, v in hw_to_id_map.items()}
+# Cache hardware instances
+_hw_id_map_cache = None
+# Declare global variables with type annotations
+hw_to_id_map: Dict[str, int] = {}
+id_to_hw_map: Dict[str, str] = {}
 
 
+def fetch_hardware_mappings():
+    global _hw_id_map_cache
+    if _hw_id_map_cache is None:
+        _hw_id_map_cache = api.get_hardware_instances()
+    return _hw_id_map_cache
```

Comment:
I don't think we should cache this for couple of reasons:
1. the hardware instance can change in the backend. So it is better to always fetch the latest list from API.
2. Even if you cache, it is not helping much, say for performance. Because the cli is executed as a new process for every single command. So no state is preserved across command execution.

Reply:
yeah I initially did caching for this because I thought this data is mostly constant and in order to reduce api calls for the user session its good to optimise the same, but yeah I agree with you we can remove the cache from here